src/transformers/modeling_utils.py

```diff
@@ -3104,155 +3017,29 @@ def from_pretrained(
-            and quantization_method_from_config is None
-        ):
-            raise ValueError(
-                "You cannot quantize with AWQ a non-quantized model using transformers, please refer to the quantization documentation"
-                " to read more about how to quantize models with AWQ algorithm https://huggingface.co/docs/transformers/main_classes/quantization"
-            )
-
-        if quantization_method_from_config is not None and quantization_method_from_args is not None:
-            if quantization_method_from_config != quantization_method_from_args:
-                raise ValueError(
-                    f"The model is already quantized with {quantization_method_from_config}. "
-                    f"You can't quantize it again with {quantization_method_from_args}"
-                )
-
-        if (
-            quantization_method_from_config in (QuantizationMethod.GPTQ, QuantizationMethod.AWQ)
-            and quantization_method_from_args is not None
-        ):
-            loading_attr_dict = quantization_config.get_loading_attributes()
-            for attr, val in loading_attr_dict.items():
-                config.quantization_config[attr] = val
-            quantization_method_from_args = None
-            logger.warning(
-                f"You passed `quantization_config` to `from_pretrained` but the model you're loading already has a "
-                f"`quantization_config` attribute and has already quantized weights. However, loading attributes"
-                f" (e.g. {list(loading_attr_dict.keys())}) will be overwritten with the one you passed to `from_pretrained`. The rest will be ignored."
-            )
-        if (
-            quantization_method_from_args == QuantizationMethod.GPTQ
-            or quantization_method_from_config == QuantizationMethod.GPTQ
-        ):
-            gptq_supports_cpu = version.parse(importlib.metadata.version("auto-gptq")) > version.parse("0.4.2")
-            if not gptq_supports_cpu and not torch.cuda.is_available():
-                raise RuntimeError("GPU is required to quantize or run quantize model.")
-            elif not (is_optimum_available() and is_auto_gptq_available()):
-                raise ImportError(
-                    "Loading a GPTQ quantized model requires optimum (`pip install optimum`) and auto-gptq library (`pip install auto-gptq`)"
-                )
-            elif version.parse(importlib.metadata.version("auto_gptq")) < version.parse("0.4.2"):
-                raise ImportError(
-                    "You need a version of auto_gptq >= 0.4.2 to use GPTQ: `pip install --upgrade auto-gptq`"
+        pre_quantized = getattr(config, "quantization_config", None) is not None
+        if pre_quantized or quantization_config is not None:
+            if pre_quantized:
+                config.quantization_config = AutoHFQuantizer.merge_quantization_configs(
+                    config.quantization_config, quantization_config
```

Comment:
it's more mergin kwargs with the config. Tokenizers has a similar logic and should be the same for every Qantizer. I don't think it should be here. It's basically passing all the kwargs to the call where you init the Config using from pretrained that it should overwrite them. (If that makes sense of course) but it's not a property of the Quantizer IMO